frontend/src/pages/SaveTemplate.tsx

```diff
@@ -55,8 +57,66 @@ export const SaveTemplate = () => {
         localStorage.setItem('templates', JSON.stringify([...existingTemplates, tempFile]))
         addFile(tempFile)
         }
+        
         navigate('/')
     }
+
+    const makeScreenshots = async () => {
+        const images: ImageData[] = JSON.parse(localStorage.getItem('images') || '[]') as ImageData[];
+        const shapes: CustomShape[][] = JSON.parse(localStorage.getItem('shapes') || '[]') as CustomShape[][];
+        const screenshots: string[] = [];
+    
+        const createScreenshot = (localImg: ImageData, shapeData: CustomShape[]) => {
+            return new Promise<string>((resolve) => {
+                const canvas = document.createElement('canvas');
+                const context = canvas.getContext('2d');
+                if (context) {
+                    canvas.width = Number(localImg.width);
+                    canvas.height = Number(localImg.height);
+                    const img = new Image();
+                    img.src = localImg.image;
+    
+                    img.onload = () => {
+                        context.drawImage(img, 0, 0, canvas.width, canvas.height);
+    
+                        shapeData.forEach((shape) => {
+                            context.fillStyle = shape.color || 'red';
+    
+                            const xCoords = shape.points.map(point => point[0]);
+                            const yCoords = shape.points.map(point => point[1]);
+    
+                            const xMin = Math.min(...xCoords);
+                            const yMin = Math.min(...yCoords);
+                            const xMax = Math.max(...xCoords);
+                            const yMax = Math.max(...yCoords);
+    
+                            const width = xMax - xMin;
+                            const height = yMax - yMin;
+    
+                            // Draw the rectangle based on the calculated coordinates
+                            context.fillRect(xMin, yMin, width, height);
+                        });
+    
+                        const screenshot = canvas.toDataURL('image/png');
```

Comment:
Do aware, this can actually have an exception of the canvas is too large. I think we likely run into this issue in other places too, it's one of the big risks with using data urls.